presto-common/src/main/java/com/facebook/presto/common/type/Uuids.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class Uuids
+{
+    private Uuids() {}
+
+    public static long[] rawUuidValuesFromBigEndian(byte[] bytes)
```

Comment:
Returning two longs is weird compared to a byte array. It's a bit of an accident that a UUID has the same number of bytes as two longs, and most importantly these aren't really numbers. Doing arithmetic on them means nothing.

Reply:
The presto implementation of the UUID type specifies that a [UUID is represented by two long values](https://github.com/prestodb/presto/blob/2f4ea64d3bf3fa8c89b3e805d5d0987ac6d971c2/presto-common/src/main/java/com/facebook/presto/common/type/UuidType.java#L156-L158). parquet's spec imposes that a UUID is a 16-byte long array. We have to convert between storage and in-memory format in order to execute on the data properly. There is no way around this.